src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,48 @@
+package core.basesyntax;
+
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+
+    private static final int QUANITY_OF_TYPES = 5;
+    private String color = ColorSupplier.getRandomColor();
+
+    public Figure getRandomFigure() {
+
+        Random random = new Random();
+        int figureType = random.nextInt(QUANITY_OF_TYPES);
+
+        switch (figureType) {
+            case 0:
+                return new Circle(color,
+                        (double) Math.round(random.nextDouble() * 100) / 100);
```

Comment:
Move each action to the private method
```suggestion
                return prepareSquare(color);
```
All private methods should be below public methods